StripeCore/StripeCore/Source/Helpers/Timeout.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  Timeout.swift
+//  StripeCore
+//
+//  Created by Joyce Qin on 10/28/25.
+//
+
+import Foundation
+
+@_spi(STP) public struct TimeoutError: Error {}
+
+// MARK: - Timeout with TaskGroup and CheckedContinuation escape hatch
+/// Runs multiple operations in parallel with individual timeouts
+/// - Parameters:
+///   - timeout: The maximum time interval to wait for each operation to complete
+///   - operations: Variadic operations to run with timeout
+/// - Returns: Tuple of Results, where each Result contains either the task's value or its error
+ @_spi(STP) public func withTimeout<each T>(
+    _ timeout: TimeInterval,
+    _ operations: repeat @escaping () async throws -> each T
+ ) async -> (repeat Result<(each T)?, Error>) {
```

Comment:
why's `(each T)?` optional? I'd think it'd be non-optional and T itself can be an optional if necessary.